library/packages/src/modules/Package.rb

```diff
@@ -221,6 +222,24 @@ def reset
       @removed_packages.clear
     end
 
+    # Tries to find a package according to the pattern
+    #
+    # @param pattern [String] a regex pattern to match, no escaping done
+    # @return list of matching package names
+    def by_pattern(pattern)
+      raise ArgumentError, "Missing search pattern" if pattern.nil? || pattern.empty?
+
+      init_packager
```

Comment:
is it safe to call it always?

Reply:
It should be responsibility of the caller to ensure that the package manager is properly initialized. It might have potentially some side effects during offline upgrade or some other less used scenarios.

The other methods here also do not run initialization so please remove it.

And for initialization you should use [PackageSystem.EnsureTargetInit](https://github.com/yast/yast-yast2/blob/1ce6c5d2f2188f757b73550e738a188bdd8c1d1c/library/packages/src/modules/PackageSystem.rb#L74) and [PackageSystem.EnsureSourceInit](https://github.com/yast/yast-yast2/blob/1ce6c5d2f2188f757b73550e738a188bdd8c1d1c/library/packages/src/modules/PackageSystem.rb#L92) calls in the caller code. These calls avoid initializing the already initialized package manager.